### PR TITLE
Pretty much a complete rewrite...

### PR DESCRIPTION
**Pull request by noplanman**
**This is not to be just merged!!**

It is more to show how I would have done certain things.

What I've changed:
- Serve only the `public` folder to the webserver, no need to allow access to all other files too. (Security)
- Temporary watermark images get saved to the `storage/images` folder, not publicly. (Security)
- Watermark is positioned 10% from top and 10% from left of image, and font size is also 10% of image height.
- Many other things that you will discover 😉 

If you really do approve of ALL these changes and understand everything completely, then I'll happily remove the `[NO MERGE]` and you can then merge it.

Sound like a plan?

P.S. It will be quite difficult to compare the changes on GitHub, because there are quite a few, so I suggest to simply get this branch and have a look at the code directly.

### CODE REVIEW

File: app/Watermark/Commands/CallbackqueryCommand.php [hunk 1]
```diff
@@ -7,13 +7,17 @@
...
+
 use Longman\TelegramBot\Commands\SystemCommand;
-use Longman\TelegramBot\Request;
+use Longman\TelegramBot\Commands\UserCommands\EditmessageCommand;
 use Longman\TelegramBot\Conversation;
-
+use Longman\TelegramBot\Entities\ServerResponse;
+use Longman\TelegramBot\Exception\TelegramException;
+use Longman\TelegramBot\Request;
 use Yii;
```
Comment (fnaysee) [inline app/Watermark/Commands/CallbackqueryCommand.php]: Remove this too !

File: app/Watermark/Commands/CallbackqueryCommand.php [hunk 2]
```diff
@@ -23,93 +27,76 @@
...
-     * @throws \Longman\TelegramBot\Exception\TelegramException
+     * @return ServerResponse
+     * @throws TelegramException
      */
-    public function execute()
+    public function execute(): ServerResponse
     {
+        $callback_data = $this->getCallbackQuery()->getData();
 
-        $this->callback_query    = $this->getCallbackQuery();
```
Comment (wp-src) [inline app/Watermark/Commands/CallbackqueryCommand.php]: Why you removed this and added it to addWatermark() as local variable ?

File: app/Watermark/Commands/GenericmessageCommand.php [hunk 3]
```diff
@@ -7,77 +7,79 @@
...
  */
+
 namespace Longman\TelegramBot\Commands\SystemCommands;
+
 use Longman\TelegramBot\Commands\SystemCommand;
 use Longman\TelegramBot\Conversation;
+use Longman\TelegramBot\Entities\ServerResponse;
+use Longman\TelegramBot\Exception\TelegramException;
 use Longman\TelegramBot\Request;
 use Yii;
```
Comment (fnaysee) [inline app/Watermark/Commands/GenericmessageCommand.php]: Remove this too

File: app/Watermark/Commands/CallbackqueryCommand.php (hunk 1 above)
Comment (noplanman) [reply app/Watermark/Commands/CallbackqueryCommand.php]: You mean you'd prefer to have it inline in the code?

Comment (fnaysee) [reply app/Watermark/Commands/CallbackqueryCommand.php]: No, i mean this line > use Yii;

**Comment by noplanman** [comment]
@wp-src Ok, free to merge now 👍

**Pull request merged**